eth/gaspricemonitor.go

```diff
@@ -70,6 +70,10 @@ func (gpm *GasPriceMonitor) SetMinGasPrice(minGasPrice *big.Int) {
 	gpm.gasPriceMu.Lock()
 	defer gpm.gasPriceMu.Unlock()
 	gpm.minGasPrice = minGasPrice
+
+	if monitor.Enabled {
```

Comment:
We'll need to record the min gas price in the GasPriceMonitor constructor as well since the min gas price can be initialized there. Otherwise, if the user never updates the min gas price while the node is running, there will be no min gas price recorded even if the node was started with the `-minGasPrice` flag.